python/ml4ir/data/tfrecord_reader.py

```diff
@@ -9,39 +9,45 @@
 from ml4ir.features.feature_config import FeatureConfig
 from ml4ir.config.keys import TFRecordTypeKey, FeatureTypeKey
 
-from typing import Union, Optional
 
 """
 This module contains helper methods for reading and writing
 data in the train.SequenceExample protobuf format
 """
 
 
-def make_parse_fn(feature_config: FeatureConfig, max_num_records: int = 25) -> tf.function:
-    """Create a parse function using the context and sequence features spec"""
+def make_parse_fn(
+    feature_config: FeatureConfig,
+    max_num_records: int = 25,
+    required_only: bool = False,
```

Comment:
@jakemannix You can now essentially pass `required_only`(parse only required features) and `pad_records`(pad records for training) as arguments to the TFRecord parse_fn.